File: conversion/Examples/HelloWorld/hello_world.ts
interface Person {
  name: string;
  age: number;
}

export function main(): void {
  console.log(`Hello, World!`);
  const sum = add(2, 2);
  console.log(`add(2, 2) = ${sum}`);
  const person = { name: "Not Sure", age: 30 };
  console.log(`Person: name=${person.name}, age=${person.age}`);
}

export function add(x: number, y: number): number {
  return x + y;
}

File: conversion/Examples/HelloWorld/hello_world.rs
fn main() {
    println!("Hello, World!");

    // Use add()
    let sum = add(2, 2);
    println!("add(2, 2) = {}", sum);

    // Allocate and use Person
    let person = Person {
        name: String::from("Not Sure"),
        age: 30,
    };
    println!("Person: name={}, age={}", person.name, person.age);
}

fn add(x: i32, y: i32) -> i32 {
    return x + y;
}

struct Person {
    name: String,
    age: u32,
}
